### PR TITLE
fix: run scheduler in a separate goroutine

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR features a simple fix that runs the scheduler in a separate goroutine; otherwise the main function will be blocked.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### CODE REVIEW

File: cmd/hubagent/workload/setup.go [hunk 1]
```diff
@@ -188,7 +188,10 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
 		defaultSchedulingQueue := queue.NewSimpleClusterResourcePlacementSchedulingQueue()
 		defaultScheduler := scheduler.NewScheduler("DefaultScheduler", defaultFramework, defaultSchedulingQueue, mgr)
 		klog.Info("Starting the scheduler")
-		defaultScheduler.Run(ctx)
+		// Scheduler must run in a separate goroutine as Run() is a blocking call.
+		go func() {
```
Comment (ryanzhang-oss) [inline cmd/hubagent/workload/setup.go:192]: when will this go routine dangle on exit? Do we not want to monitor/wait for it?

Comment (michaelawyu) [reply cmd/hubagent/workload/setup.go:192]: Hi Ryan! The scheduler is set to exit on context cancellation (same for the controller manager).

Comment (michaelawyu) [reply cmd/hubagent/workload/setup.go:192]: I will add a waitgroup to make things clearer for both.

**Review by zhiying-lin — APPROVED** [review]

File: cmd/hubagent/main.go [hunk 2]
```diff
@@ -42,6 +43,8 @@ var (
 		handleExitFunc()
 		os.Exit(1)
 	}
+
+	wg sync.WaitGroup
```
Comment (zhiying-lin) [inline cmd/hubagent/main.go:47]: declare the wg in the main func instead? otherwise, the whole main package can update the wg.

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**